cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -97,6 +97,7 @@ public RepositoryManager(final SecureStore secureStore,
       // This is not expected as only valid auth configs will be stored.
       throw new RuntimeException(e);
     }
+    SecureSystemReader.setAsSystemReader();
```

Comment:
It's not good for constructors to have global side effects like this. It's not clear to caller that creating a new instance will impact the behavior of the rest of the jvm and can easily lead to strange bugs.

Global static things should be setup once when starting up. For example, add it as one of the first things done in the SourceControlTask. May need to refactor some classes like the InMemorySourceControlOperationRunner to extend AbstractIdleService and have a startUp() method.

Reply:
Made changes so that the `SystemReader` is set at startup in Appfabric and in Task Workers.